examples/game/lib/game_demo_node.dart

```diff
@@ -181,21 +183,24 @@ class GameDemoNode extends NodeWithSize {
     if (chunk == 0) {
       // Leave the first chunk empty
       return;
-    } else if (chunk == 1) {
-      addLevelAsteroids(10, yPos, 0.0);
-    } else {
-      addLevelAsteroids(9 + chunk, yPos, 0.5);
     }
-  }
 
-  void addLevelAsteroids(int numAsteroids, double yPos, double distribution) {
-    for (int i = 0; i < numAsteroids; i++) {
-      GameObjectType type = (randomDouble() < distribution) ? GameObjectType.asteroidBig : GameObjectType.asteroidSmall;
-      Point pos = new Point(randomSignedDouble() * 160.0,
-                            yPos + _chunkSpacing * randomDouble());
-      _objectFactory.addGameObject(type, pos);
+    chunk -= 1;
+
+    int level = chunk ~/ _chunksPerLevel;
+    int part = chunk % _chunksPerLevel;
+
+    if (part == 0) {
+      _objectFactory.addAsteroids(10 + level * 4, yPos, 0.0 + (level * 0.2).clamp(0.0, 0.7));
```

Comment:
This is a bit of work in progress, once I have more things to fill the game with there will be more variation. :)